src/core/include/openvino/core/except.hpp

```diff
@@ -45,6 +45,13 @@ class OPENVINO_API AssertFailure : public Exception {
                                  const std::string& context_info,
                                  const std::string& explanation);
 };
+
+/// Exception class to be thrown on not implemented code
+class OPENVINO_API NoImpl : public AssertFailure {
```

Comment:
Why not `NotImplemented`?

Reply:
But why we have wrong type? We use different namespaces it should solve the issue with intersected names.

Reply:
Added `InferneceEngine::` namespace everywhere